### PR TITLE
Updating core tools generator to use metadata.json in the output

### PR DESCRIPTION
**Pull request by soninaren**

### CODE REVIEW

File: GenerateToolingFeed/Helper.cs [hunk 1]
```diff
@@ -194,5 +170,28 @@ public static string GetReleaseVersionFromTag(JObject json, string tag)
             4 => ["v4", "v0"],
             _ => throw new ArgumentNullException($"{majorVersion} is not a supported version.", nameof(majorVersion))
         };
+
+        public static string GetCliVersion(string path)
```
Comment (aishwaryabh) [inline GenerateToolingFeed/Helper.cs:174]: do we still need this logic if we are just reading from metadata.json file?

File: GenerateToolingFeed/V3Format/V3FormatFeedEntryUpdater.cs [hunk 2]
```diff
@@ -59,5 +59,20 @@ private static bool ShouldBeMinified(V3FormatCliEntry cliEntry)
             return os.Equals("Windows", StringComparison.OrdinalIgnoreCase)
                 && cliEntry.Architecture.Equals("x64", StringComparison.OrdinalIgnoreCase);
         }
+
+        private static string GetDownloadLink(string os, string architecture, string cliVersion, bool isMinified = false, string linkSuffix = "")
```
Comment (aishwaryabh) [inline GenerateToolingFeed/V3Format/V3FormatFeedEntryUpdater.cs:63]: I think linkSuffix would be `inproc` right?

Comment (soninaren) [reply GenerateToolingFeed/V3Format/V3FormatFeedEntryUpdater.cs:63]: yeah, its passed in the caller

File: GenerateToolingFeed/Helper.cs (hunk 1 above)
Comment (soninaren) [reply GenerateToolingFeed/Helper.cs:174]: This is for feed v3 in case we need to generate it.

**Review by aishwaryabh — APPROVED** [review]

**Pull request merged**